### PR TITLE
Feature/test txconflictregistry anthony

### PR DESCRIPTION
**Pull request by anthonyromaniello**
Summary
Implemented comprehensive test suite for TxConflictRegistry class with test design documentation.

Changes
- **Test Design Document**: Created complete test design with black box and white box analysis for all methods
- **Test Implementation**: 39 test cases covering:
  - Constructor (7 tests)
  - setMatch (10 tests)
  - noMatch (9 tests)
  - neutralize (4 tests)
  - getMatch (5 tests)
  - uninitialized (4 tests)
- **Bug Fixes**: Fixed 16 test implementation issues including assertion patterns and boundary conditions

Testing
- All 39 tests passing
- Full test suite: 245 tests, 0 failures, 0 errors
- 100% method coverage for TxConflictRegistry (minus toString: assumption that it wasn't necessary since it is a display method)

Commits
- docs(transaction): Add Javadocs for TxConflictRegistry methods
- docs(transaction): Add test design for TxConflictRegistry
- test(transaction): Implement and fix TxConflictRegistry test suite

### CODE REVIEW

File: src/test/resources/test-design/Transaction/TxConflictRegistryTest.md [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+## 1. TxConflictRegistry(long size) --> Constructor
+
+### Black Box Analysis
+
+#### Input partitions
+| ID | Condition | Partition | Valid? |
+|----|-----------|-----------|--------|
+| P1 | size value | size = 1 | Valid |
+| P2 | size value | 1 < size <= Integer.MAX_VALUE | Valid |
+| P3 | size value | size < 1 | Invalid |
```
Comment (cmg-york) [inline src/test/resources/test-design/Transaction/TxConflictRegistryTest.md:13]: Specify precise behavior (what exception it throws)

File: src/test/resources/test-design/Transaction/TxConflictRegistryTest.md [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+| P4 | size value | size > Integer.MAX_VALUE | Invalid |
+
+#### Boundary values
+| ID | Variable-Value |
```
Comment (cmg-york) [inline src/test/resources/test-design/Transaction/TxConflictRegistryTest.md:17]: Valid? column

File: src/test/resources/test-design/Transaction/TxConflictRegistryTest.md [hunk 3]
```diff
@@ -0,0 +1,414 @@
...
+|----|----------------|
+| B1 | size = 1 (minimum valid) |
+| B2 | size = 2 (just above minimum) |
+| B3 | size = Integer.MAX_VALUE (maximum valid) |
+| B4 | size = 0 (just below valid) |
+| B5 | size = -1 (negative value) |
+| B6 | size = Integer.MAX_VALUE + 1L (just above valid) |
+
+#### Guesses
+| ID | Variable-Value |
```
Comment (cmg-york) [inline src/test/resources/test-design/Transaction/TxConflictRegistryTest.md:27]: Valid? column?

File: src/test/resources/test-design/Transaction/TxConflictRegistryTest.md [hunk 4]
```diff
@@ -0,0 +1,414 @@
...
+## 2. setMatch(int a, int b)
+
+### Black Box Analysis
+
+#### Input partitions
+| ID | Condition | Partition | Valid? |
+|----|-----------|-----------|--------|
+| P1 | a value | a = 1 | true |
+| P2 | a value | a > 1 | true |
+| P3 | a value | a <= 0 | false |
```
Comment (cmg-york) [inline src/test/resources/test-design/Transaction/TxConflictRegistryTest.md:69]: Describe exact behavior (exception)

**Review by cmg-york — COMMENTED** [review]
1. Refer to the tests from your test plan from the *Test.java file (in comments). In addition, include more comments on the test methods that describe what test tests in words.
2. Remove log, .project from the commit.
3. Address comments in MD
4. Is this in co-ordination with Edison?
5. [this can be separate commit] Please prepare test cases and tests that combine, setMatch, getMatch and noMatch.

File: src/main/java/ca/yorku/cmg/cnsim/engine/transaction/TxConflictRegistry.java [hunk 5]
```diff
@@ -4,13 +4,23 @@
 
 public class TxConflictRegistry {
```
Comment (cmg-york) [inline src/main/java/ca/yorku/cmg/cnsim/engine/transaction/TxConflictRegistry.java:5]: Anthony, Edison: please add JML contracts to all methods, such that they also appear on JavaDoc.

**Review by cmg-york — COMMENTED** [review]
Hello again Anthony: please see additional comment on JML contracts.
